### ISSUE TITLE
Distinguish between deprecated globals and the rest

### ISSUE THREAD

**Issue by GrosSacASac** [eslint/eslint#12167]
**What rule do you want to change?**

"no-shadow" with "builtinGlobals" from eslint 6.2.2

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

The new default should not warn if a global deprecated variable is overshadowed.

"no-shadow": [1, { "builtinGlobals": true, includeDeprecated: false}],
"no-shadow": [1, { "builtinGlobals": true, includeDeprecated: true }],

**Please provide some example code that this change will affect:**

```js
button.addEventListener("click", function (event) {

}):
```

**What does the rule currently do for this code?**

'event' is already declared in the upper scope

**What will the rule do after it's changed?**

nothing

**Are you willing to submit a pull request to implement this change?**

no

**Comment by platinumazure** [comment]
Hi @GrosSacASac, thanks for the issue.

Just out of curiosity, can this also be resolved by specifying `"event": "off"` in your globals configuration?

**Comment by GrosSacASac** [comment]
There are others as well

**Comment by platinumazure** [comment]
Sure, what I'm getting at is you or someone could create a new environment or shareable configuration with the necessary deprecated globals turned off. :smile:
